iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/gcp/GcpPrivilegePolicyCheckPart.java

```diff
@@ -0,0 +1,123 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.terraform.checks.gcp;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.common.api.checks.SecondaryLocation;
+import org.sonar.iac.common.checks.PropertyUtils;
+import org.sonar.iac.common.extension.visitors.TreeContext;
+import org.sonar.iac.common.extension.visitors.TreeVisitor;
+import org.sonar.iac.terraform.api.tree.AttributeAccessTree;
+import org.sonar.iac.terraform.api.tree.AttributeTree;
+import org.sonar.iac.terraform.api.tree.BlockTree;
+import org.sonar.iac.terraform.api.tree.FileTree;
+import org.sonar.iac.terraform.api.tree.TerraformTree;
+import org.sonar.iac.terraform.checks.AbstractNewResourceCheck;
+import org.sonar.iac.terraform.checks.utils.TerraformUtils;
+import org.sonar.iac.terraform.symbols.ResourceSymbol;
+
+import static org.sonar.iac.terraform.checks.AbstractResourceCheck.getResourceType;
+import static org.sonar.iac.terraform.checks.AbstractResourceCheck.isResource;
+
+public class GcpPrivilegePolicyCheckPart extends AbstractNewResourceCheck {
+
+  private static final String POLICY_MESSAGE = "Make sure it is safe to give all members full access.";
+  private static final String MEMBER_MESSAGE = "Make sure it is safe to grant that member full access.";
+  private static final String SECONDARY_MESSAGE = "The policy is used here.";
+
+  private final Map<String, AttributeTree> policyReferences = new HashMap<>();
+
+  @Override
+  public void initialize(InitContext init) {
+    super.initialize(init);
+    init.register(FileTree.class, (ctx, tree) -> new PolicyReferenceCollector().scan(new TreeContext(), tree));
+  }
+
+  @Override
+  protected void provideResource(CheckContext ctx, BlockTree blockTree) {
+    super.provideResource(ctx, blockTree);
+    if (isData(blockTree)) {
+      ResourceSymbol data = ResourceSymbol.fromPresent(ctx, blockTree);
+      if ("google_iam_policy".equals(data.type)) {
+        checkPolicy(data);
+      }
+    }
+  }
+
+  private static boolean isData(BlockTree blockTree) {
+    return "data".equals(blockTree.key().value());
+  }
+
+  private void checkPolicy(ResourceSymbol data) {
+    AttributeTree reference = policyReferences.get(formatReference(data));
+    if (reference != null) {
+      SecondaryLocation secondary = new SecondaryLocation(reference, SECONDARY_MESSAGE);
+      data.blocks("binding").forEach(
+        binding -> binding.attribute("role")
+          .reportIf(matchesPattern(".*(?:admin|manager|owner|superuser).*"), POLICY_MESSAGE, secondary));
```

Comment:
Since it has to contain one of these 4 words, I'd suggest this edition of the regex:
`".*\b(?:admin|manager|owner|superuser)\b.*"`
Otherwise `managerial` and `administrative` will also be matched.

Reply:
Yeah. So far I've got the ticket as well as the [documentation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_project_iam#role) we would like to catch all roles containing the sensitive words.
For example, there could be a custom role Superadmin without a dot. I would say the number of FPs is less than the number of FNs when we would change this.